shell/common/rasterizer.cc

```diff
@@ -264,9 +267,31 @@ RasterStatus Rasterizer::DoDraw(
   // TODO(liyuqian): in Fuchsia, the rasterization doesn't finish when
   // Rasterizer::DoDraw finishes. Future work is needed to adapt the timestamp
   // for Fuchsia to capture SceneUpdateContext::ExecutePaintTasks.
-  timing.Set(FrameTiming::kRasterFinish, fml::TimePoint::Now());
+  const auto raster_finish_time = fml::TimePoint::Now();
+  timing.Set(FrameTiming::kRasterFinish, raster_finish_time);
   delegate_.OnFrameRasterized(timing);
 
+  const fml::TimePoint latest_frame_target_time =
+      delegate_.GetLatestFrameTargetTime();
+  if (latest_frame_target_time > frame_target_time) {
+    const auto frame_budget_millis = delegate_.GetFrameBudget().count();
+    const int vsync_transitions_missed =
+        (latest_frame_target_time - latest_frame_target_time).ToMilliseconds() /
+        frame_budget_millis;
```

Comment:
An integer division may round down. Note that the number of millis is already rounded in both cases so you may be 1 under the right number of frames.

Would it make sense to count the "current frame number" in the delegate_? If not, then using a rounding calculating - add half of the frame_budget_millis to the difference before dividing...

Reply:
Makes sense, will round them.